File: Greatest-element-of-a-list/Java/greatest-element-of-a-list-1.java
import java.util.NoSuchElementException;

class Greatest{

	public static float max(float[] values) throws NoSuchElementException {
	    if (values.length == 0)
		throw new NoSuchElementException();
	    float themax = values[0];
	    for (int idx = 1; idx < values.length; ++idx) {
		if (values[idx] > themax)
		    themax = values[idx];
	    }
	    return themax;
	}

	public static void main(String[] args){
		float values[] = {1,2,3,1,4,2,1};
		System.out.println(max(values));
	}

}

File: Greatest-element-of-a-list/C/greatest-element-of-a-list-1.c
#include <assert.h>
#include <stdio.h>

float max(unsigned int count, float values[]) {
     assert(count > 0);
     //unsigned int idx;
     int i;
     float themax = values[0];
     for(i = 1; i < count; ++i) {
          themax = values[i] > themax ? values[i] : themax;
     }
     return themax;
}

int main(){
	float values[] = {1,2,3,1,4,2,1};
	float maxvalue = max(7, values);
	printf("%f\n", maxvalue);
	return 0;
}
